### ISSUE TITLE
Use ConnectionFactoryWrapperBuildItem in Quarkus

### ISSUE THREAD

**Issue by zhfeng** [quarkiverse/quarkus-artemis#148]
We need to use `ConnectionFactoryWrapperBuildItem` instead of  `ArtemisJmsWrapperBuildItem` since https://github.com/quarkusio/quarkus/pull/29753 get merged.

And I also create
- https://github.com/quarkiverse/quarkus-pooled-jms/issues/71

**Comment by zhfeng** [comment]
@turing85 yeah, I'm working on it and please assign to me.

**Comment by turing85** [comment]
@zhfeng eager to see your implementation. I just tried it myself, but for quarkus `2.16.2.Final`, I have - for some reason, define the version of `quarkus-jms-spi-deployment` explicitly, which bothers me somewhat.

**Comment by zhfeng** [comment]
@turing85 Hmm, `quarkus-jms-spi-deployment` is in the `io.quarkus:quarkus-bom`?

**Comment by turing85** [comment]
@zhfeng Nope. Looks like it should be added [here](https://github.com/quarkusio/quarkus/blob/HEAD/bom/application/pom.xml#L613), but it is not present in this file.

**Comment by zhfeng** [comment]
OK, I see and I can understand it is not there because `quarkus-jms-spi-deployment` can only be used by extensions such as `quarkus-artemis` and `quarkus-pooled-jms`. The application does not need it. Then we have to define version.

**Comment by zhfeng** [comment]
@turing85 is there any other issue when you tried?

**Comment by turing85** [comment]
@zhfeng yes. The new build item is only accessible in the `deployment` module. I am not yet entirely sure how to translate all this. I think we want to keep the `ArtemisJmsWrapper`-class, and only transform this "at the last moment".

**Comment by zhfeng** [comment]
@turing85 no problem and I just raise a PR for discussing. I can not catch your idea about  "only transform this  at the last moment" which mean only have it within quarkus 3 support?

**Comment by turing85** [comment]
@zhfeng so my idea was to basically keep everything "as-is", and just add a `BuildProducer<ConnectionFactoryWrapperBuildItem> connectionFactoryWrapperBuilder` to the parameter-list of`ArtemisJmsProcessor::configure` and create a `ConnectionFactoryWrapperBuildItem` before `return`ing.

**Comment by zhfeng** [comment]
@turing85 I confused about your idea. It seems that we only need to consume `ConnectionFactoryWrapperBuildItem` but not produce it. On the other hand `quarkus-pooled-jms` produces it.

**Comment by turing85** [comment]
@zhfeng OOOOOH! Okay. That was the reason I was unable to find a stage that produces the `QuarkusArtemisWrapperBuildItem` :smile: That makes it much, much simpler.

**Comment by turing85** [comment]
> OK, I see and I can understand it is not there because `quarkus-jms-spi-deployment` can only be used by extensions such as `quarkus-artemis` and `quarkus-pooled-jms`. The application does not need it. Then we have to define version.

I am not sure I agree. Are there other extensions that are not loaded by default? It is "just a bom", it should be in there. We would never want a version of that extension that differs from the quarkus version.

**Comment by turing85** [comment]
I opened a in quarkus to add it to the bom: https://github.com/quarkusio/quarkus/pull/31074

**Comment by zhfeng** [comment]
Thanks for raising these PRs!
